### ISSUE TITLE
Mfrc522 Rfid Reader

### ISSUE THREAD

**Issue by Khalifla** [wokwi/wokwi-features#359]
It would help simulate RFID projects.

**Comment by urish** [comment]
Thanks! Opened this request for [voting](https://wokwi.com/features?sort=newest#feature-1242782423)

**Comment by martinius96** [comment]
Bro work on it. 9 months later and no progress

**Comment by urish** [comment]
Sure, just make sure it gets to the [top of the list](https://wokwi.com/features) first

**Comment by martinius96** [comment]
Yep boss. For that 1100 points top "feature" hasn't been implemented for more than year, when it was on first position. 
Let people send you more money and maybe at 10000 points u start doing on it.

**Comment by urish** [comment]
Maybe :)

**Comment by untillnesss** [comment]
+1

**Comment by leftCoast** [comment]
Looks like you were priced out of the market. That's life.

-jim lee

**Comment by MarioRogue** [comment]
Is that actually going to happen ?

**Comment by martinius96** [comment]
No, because u need to pay at least 500 USD to have any chance of implementing this feature in next year or two.

**Comment by Kujbor** [comment]
+1

**Comment by AbdennourBELKACEM** [comment]
come on guy ,we neeed the rfid module ,  after all this time with no progress ,i think the dev guys are sleeping

**Comment by martinius96** [comment]
From time to time I browse Google for RFID Wokwi projects.
There are some, also compilation is running, custom module for RFID RC522 is there, but I don't know how to use it.
There is no guide on "How to attach a tag". Maybe it works in some way that I haven't found yet.

Examples I found:
https://wokwi.com/projects/389180983211282433

https://wokwi.com/projects/395416055497496577

**Comment by djedu28** [comment]
Hi @martinius96, I reviewed the links you presented, none of these projects have functional implementation of the chip.

1. The first link started implementing, but didn't complete the SPI communication on the chip. Maybe someone can help with that. Mfrc522 uses SPI communication. Here's the documentation for customizing chips with SPI: https://docs.wokwi.com/chips-api/spi

2. The second link, the chip code doesn't do anything.

-----

As an alternative to `CHIP RC522`, I made my own [`CHIP RDM6300`](https://github.com/djedu28/wokwi-rdm6300-custom-chip/blob/main/docs/README.md), you can use it

> In the past I had the need to emulate an RFID reader, and I did a basic simulation of the RDM6300 Chip https://github.com/djedu28/wokwi-rdm6300-custom-chip
> [![image](https://github.com/user-attachments/assets/1b3b756a-39f6-489d-a54a-e53e4505f77c)](https://github.com/djedu28/wokwi-rdm6300-custom-chip/blob/main/docs/README.md)
  
> The RDM6300 chip uses serial communication.
> I implemented 3 controls. 
> ![image](https://github.com/user-attachments/assets/59b15975-65c1-4ffa-8750-db022cf6f403)
>
> * Control-1 > "1-Aproximate" or "0-Away" the tag
> * Control-2 > Card1: "1-On" and "0-Off" tag 1
> * Control-3 > Card2: "1-On" and "0-Off" tag 2
 
> When Card1=0 and Card2=1, the chip approximates or moves away tag 2 (Card2), any other configuration uses tag 1


> Automatically translated.
-----
Original message (pt-BR):

Olá @martinius96, revisei os links que você apresentou, nenhum destes projetos possuem a implementação funcional do chip
>
>1. O primeiro link iniciou a implementação, contudo não foi realizado a comunicação SPI no chip. Talvez alguém possa dar comunidade. A comunicação do Mfrc522 é SPI. Aqui possui a documentação da personalização de chips com SPI <https://docs.wokwi.com/chips-api/spi>
>
>2. O segundo link, o código do chip não faz nada.
>
>-----
>
>Como alternativa ao `CHIP RC522`, fiz o meu próprio [`CHIP RDM6300`](https://github.com/djedu28/wokwi-rdm6300-custom-chip/blob/main/docs/README.md), você pode usá-lo
>
>> No passado tive a necessidade de emular um leitor de RFID, e fiz uma simulação básica do RDM6300 Chip <https://github.com/djedu28/wokwi-rdm6300-custom-chip>
>> ![image](https://github.com/user-attachments/assets/1b3b756a-39f6-489d-a54a-e53e4505f77c)
>> 
>> o chip RDM6300  usa comunicação serial.
>> implementei 3 controles.
>> [![image](https://github.com/user-attachments/assets/59b15975-65c1-4ffa-8750-db022cf6f403)](https://github.com/djedu28/wokwi-rdm6300-custom-chip/blob/main/docs/README.md)
>>
>> * Controle-1 >  "1-aproximar" ou "0-Afastar" a tag
>> * Controle-2 >  Card1: "1-Ativa" e "0-Desativa" a tag 1
>> * Controle-3 >  Card2: "1-Ativa" e "0-Desativa" a tag 2
>
>> Quando a Card1=0 e a Card2=1, o chip aproxima ou afasta a tag 2 (Card2), qualquer outra configuração usa a tag1
>

**Comment by urish** [comment]
There we go, the MRFC522 is now officially supported on Wokwi!

Docs: https://docs.wokwi.com/parts/board-mfrc522
Example project: https://wokwi.com/projects/454873133691473921

**Comment by martinius96** [comment]
Works correctly, even with ATmega328P AVR chips. 
It was worth the wait. PN532 soon? :)

**Comment by urish** [comment]
For PN532 , feel free to open a separate issue
